packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -0,0 +1,146 @@
+
+    constructor(
+        private readonly logger: ITelemetryLogger,
+        private readonly getAuthorizationHeader: AuthorizationHeaderGetter,
+        baseurl?: string,
+        defaultQueryString: Record<string, unknown> = {},
+    ) {
+        super(baseurl, defaultQueryString);
+    }
+
+    public async load() {
+        this.authorizationHeader = await this.getAuthorizationHeader();
+    }
+
+    protected async request<T>(requestConfig: AxiosRequestConfig, statusCode: number, canRetry = true): Promise<T> {
+        const config = {
+            ...requestConfig,
+            headers: this.generateHeaders(requestConfig.headers),
+        };
+
+        try {
+            const response = await Axios.request<T>(config);
+            return response.data;
+        } catch (reason: any) {
+            if (!reason || !reason?.isAxiosError) {
+                // Unknown error, treat as critical error and immediately throw as non-retriable
+                this.logger.sendErrorEvent({
+                    eventName: "CriticalRequestError",
+                    correlationId: config.headers["x-correlation-id"] as string,
+                }, reason);
+                throwR11sNetworkError(`Unknown Error on [${config.method}] to [${config.url}]: ${
+                    safeStringify(reason)
+                }`);
+            }
+
+            const axiosError = reason as AxiosError;
+            if (axiosError.response?.status === statusCode) {
+                // Axios misinterpreted as error, return as successful response
+                return axiosError.response.data as T;
+            }
+
+            if (axiosError.response?.status === 401 && canRetry) {
+                // Refresh Authorization header and retry once
+                this.authorizationHeader = await this.getAuthorizationHeader(true);
+                return this.request<T>(config, statusCode, false);
+            }
+
+            if (axiosError.response?.status === 429 && axiosError.response.data?.retryAfter > 0) {
+                // Retry based on retryAfter[Seconds]
+                return new Promise<T>((resolve, reject) => setTimeout(() => {
+                    this.request<T>(config, statusCode)
+                        .then(resolve)
+                        .catch(reject);
+                }, axiosError.response!.data.retryAfter * 1000));
+            }
+
+            // Allow anything else to be handled upstream
+            throwR11sNetworkError(axiosError.message, axiosError.response?.status);
+        }
+    }
+
+    private generateHeaders(requestHeaders?: Record<string, unknown>): Record<string, unknown> {
+        const correlationId = requestHeaders?.["x-correlation-id"] || uuid();
+
+        return {
+            "x-correlation-id": correlationId,
+            "Authorization": this.authorizationHeader,
+            ...requestHeaders,
+        };
+    }
+}
+
+export class RouterliciousStorageRestWrapper extends RouterliciousRestWrapper {
```

Comment:
Should we make these two class static? And call load() to return the constructor?